go-runtime/modulecontext/module_context.go

```diff
@@ -2,59 +2,206 @@ package modulecontext
-	cm, err := cf.NewDefaultConfigurationManagerFromEnvironment(ctx)
+// "value" must be a pointer to a Go type that can be unmarshalled from JSON.
+func (m *ModuleContext) GetConfig(name string, value any) error {
+	data, ok := m.configs[name]
+	if !ok {
+		return fmt.Errorf("no config value for %q", name)
+	}
+	return json.Unmarshal(data, value)
+}
+
+// SetConfig sets a configuration value for the module.
+func (m *ModuleContext) SetConfig(name string, value any) error {
+	data, err := json.Marshal(value)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	sm, err := cf.NewDefaultSecretsManagerFromEnvironment(ctx)
+	m.SetConfigData(name, data)
+	return nil
+}
+
+// SetConfigData sets a configuration value with raw bytes
+func (m *ModuleContext) SetConfigData(name string, data []byte) {
+	m.configs[name] = data
+}
+
+// GetSecret reads a secret value for the module.
+//
+// "value" must be a pointer to a Go type that can be unmarshalled from JSON.
+func (m *ModuleContext) GetSecret(name string, value any) error {
+	data, ok := m.secrets[name]
+	if !ok {
+		return fmt.Errorf("no secret value for %q", name)
+	}
+	return json.Unmarshal(data, value)
+}
+
+// SetSecret sets a secret value for the module.
+func (m *ModuleContext) SetSecret(name string, value any) error {
+	data, err := json.Marshal(value)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	dbp, err := NewDBProviderFromEnvironment(module)
+	m.SetSecretData(name, data)
+	return nil
+}
+
+// SetSecretData sets a secret value with raw bytes
+func (m *ModuleContext) SetSecretData(name string, data []byte) {
+	m.secrets[name] = data
+}
+
+// AddDatabase adds a database connection
+func (m *ModuleContext) AddDatabase(name string, dbType DBType, dsn string) error {
+	db, err := sql.Open("pgx", dsn)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return &ModuleContext{
-		module:         module,
-		configManager:  cm,
-		secretsManager: sm,
-		dbProvider:     dbp,
-	}, nil
+	m.databases[name] = dbEntry{
+		dsn:    dsn,
+		db:     db,
+		dbType: dbType,
+	}
+	return nil
 }
+
+// GetDatabase gets a database connection
+//
+// Returns an error if no database with that name is found or it is not the expected type
+func (m *ModuleContext) GetDatabase(name string, dbType DBType) (*sql.DB, error) {
+	entry, ok := m.databases[name]
+	if !ok {
+		return nil, fmt.Errorf("missing DSN for database %s", name)
+	}
+	if entry.dbType != dbType {
+		return nil, fmt.Errorf("database %s does not match expected type of %s", name, dbType)
+	}
+	return entry.db, nil
+}
+
+// BehaviorForVerb returns what to do to execute a verb
+//
+// This allows module context to dictate behavior based on testing options
+func (m *ModuleContext) BehaviorForVerb(ref Ref) (VerbBehavior, error) {
+	if mock, ok := m.mockVerbs[ref]; ok {
+		return MockBehavior{Mock: mock}, nil
+	}
+	// TODO: add logic here for when to do direct behavior
```

Comment:
Will be done in this PR after it rebases on this one: https://github.com/TBD54566975/ftl/pull/1373